R/db.R

```diff
@@ -96,8 +96,10 @@ dbWriteTableUntilSuccess <- function(dbconn, tblname, df, append = FALSE, row.na
 
   repeat {
     class_map <- list(integer = 'bigint', numeric = 'double precision', factor = 'text',
-                      double = 'double precision', character = 'text', logical = 'text')
-    field_types <- sapply(sapply(df, class), function(klass) class_map[[klass]])
+                      double = 'double precision', character = 'text', logical = 'text',
+                      POSIXct = 'timestamp')
+    field_classes <- sapply(df, function(col) head(class(col), 1))
```

Comment:
Please use `class(col)[1L]`. Even if you went with head, you'd have to do `utils::head`.